src/main/java/org/opensearch/jobscheduler/model/ExtensionJobParameter.java

```diff
@@ -0,0 +1,160 @@
+        this.lockDurationSeconds = lockDurationSeconds;
+        this.jitter = jitter;
+    }
+
+    public ExtensionJobParameter(ScheduledJobParameter jobParameter) {
+        // Convert job Parameter into writeable ExtensionJobParameter
+        this(
+            jobParameter.getName(),
+            jobParameter.getSchedule(),
+            jobParameter.getLastUpdateTime(),
+            jobParameter.getEnabledTime(),
+            jobParameter.isEnabled(),
+            jobParameter.getLockDurationSeconds(),
+            jobParameter.getJitter()
+        );
+    }
+
+    public ExtensionJobParameter(StreamInput in) throws IOException {
+        this.jobName = in.readString();
+        if (in.readEnum(ExtensionJobParameter.ScheduleType.class) == ScheduleType.CRON) {
+            this.schedule = new CronSchedule(in);
+        } else {
+            this.schedule = new IntervalSchedule(in);
+        }
+        this.lastUpdateTime = in.readInstant();
+        this.enabledTime = in.readInstant();
+        this.isEnabled = in.readBoolean();
+        this.lockDurationSeconds = in.readOptionalLong();
+        this.jitter = in.readOptionalDouble();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(this.jobName);
+        if (this.schedule instanceof CronSchedule) {
+            out.writeEnum(ScheduleType.CRON);
+        } else {
+            out.writeEnum(ScheduleType.INTERVAL);
+        }
+        this.schedule.writeTo(out);
+        out.writeInstant(this.lastUpdateTime);
+        out.writeInstant(this.enabledTime);
+        out.writeBoolean(this.isEnabled);
+        out.writeOptionalLong(this.lockDurationSeconds);
+        out.writeOptionalDouble(this.jitter);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(NAME_FIELD, this.jobName)
+            .field(SCHEDULE_FIELD, this.schedule)
+            .field(LAST_UPDATE_TIME_FIELD, lastUpdateTime.toEpochMilli())
+            .field(ENABLED_TIME_FIELD, enabledTime.toEpochMilli())
+            .field(IS_ENABLED_FIELD, isEnabled);
+        if (this.lockDurationSeconds != null) {
+            builder.field(LOCK_DURATION_SECONDS_FIELD, this.lockDurationSeconds);
+        }
+        if (this.jitter != null) {
+            builder.field(JITTER_FIELD, this.jitter);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public String getName() {
+        return this.jobName;
+    }
+
+    @Override
+    public Instant getLastUpdateTime() {
+        return this.lastUpdateTime;
+    }
+
+    @Override
+    public Instant getEnabledTime() {
+        return this.enabledTime;
+    }
+
+    @Override
+    public Schedule getSchedule() {
+        return this.schedule;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return this.isEnabled;
+    }
+
```

Comment:
Do you want getters for the durationSeconds and jitter fields?   (Could document nullable or make Optionals.)

Reply:
Great catch, unfortunately I missed adding those, I'll go ahead and do so.